cmd/smoke-test/smoketest.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	log "github.com/golang/glog"
+	"github.com/golang/protobuf/proto"
+	"github.com/onosproject/onos-config/pkg/utils"
+	"github.com/openconfig/gnmi/client"
+	gclient "github.com/openconfig/gnmi/client/gnmi"
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+const (
+	tzValue = "Europe/Dublin"
+	tzPath = "/system/clock/config/timezone-name"
+	device = "device-1-device-simulator"
+)
+var (
+	clientKeyPath = "/Users/ray/go/src/github.com/onosproject/onos-config/deployments/helm/onos-config/files/certs/tls.key"
+	clientCrtPath = "/Users/ray/go/src/github.com/onosproject/onos-config/deployments/helm/onos-config/files/certs/tls.crt"
+	caCertPath    = "/Users/ray/go/src/github.com/onosproject/onos-config/deployments/helm/onos-config/files/certs/tls.cacrt"
+	host          = "config.onosproject.org"
+	port          = "443"
+	address       = host + ":" + port
+)
+
+func makePathProto(target string, path string) string {
+	var protoBuilder strings.Builder
+
+	protoBuilder.WriteString("path: <target: '")
+	protoBuilder.WriteString(target)
+	protoBuilder.WriteString("', ")
+
+	pathElements := utils.SplitPath(path)
+	for _, pathElement := range pathElements {
+		protoBuilder.WriteString("elem: <name: '")
+		protoBuilder.WriteString(pathElement)
+		protoBuilder.WriteString("'> ")
+	}
+	protoBuilder.WriteString(">")
+	return protoBuilder.String()
+}
+
+func makeSetPathProto(target string, path string, value string) string {
+	var protoBuilder strings.Builder
+
+	protoBuilder.WriteString("update: <path: <target: '")
+	protoBuilder.WriteString(target)
+	protoBuilder.WriteString("', ")
+
+	pathElements := utils.SplitPath(path)
+	for _, pathElement := range pathElements {
+		protoBuilder.WriteString("elem: <name: '")
+		protoBuilder.WriteString(pathElement)
+		protoBuilder.WriteString("'> ")
+	}
+	protoBuilder.WriteString(">")
+	protoBuilder.WriteString(" val: <string_val: '")
+	protoBuilder.WriteString(value)
+	protoBuilder.WriteString("'>>")
+	return protoBuilder.String()
+}
+
+func findPathValue(response *gpb.GetResponse, path string) (string, error) {
+	if len(response.Notification) != 1 {
+		return "", errors.New("response notifications must have one entry")
```

Comment:
just out of my curiosity, why not fmt.Errorf ?

Reply:
It should be. That seems to be the standard.

Reply:
done